compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -955,8 +955,22 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                             );
                         }
 
-                        let body_hir_id =
-                            self.tcx.hir().local_def_id_to_hir_id(obligation.cause.body_id);
+                        let body_def_id = if let Some(rpitit_info) =
```

Comment:
I think you will eventually remove this, see comment below. But in general, hacks like this need a comment explaining why they're needed and why they're valid.